{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from itertools import islice\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "class Kmer_spectrum:\n",
    "    ### This class represents kmer_spectrum of a sequence, kmers have fixed size. ###\n",
    "    counter = 0\n",
    "    \n",
    "    def __init__(self,kmer_name):\n",
    "        self.sequence = kmer_name\n",
    "        self.loci = [] # list of found loci coordinates\n",
    "        self.counter = 1 # the number of appearance of kmer in string\n",
    "\n",
    "    def increase(self):\n",
    "        self.counter += 1\n",
    "        \n",
    "    def increase_n(self, n):\n",
    "        self.counter += n\n",
    "        \n",
    "    def add_locus(self, chrom, position):\n",
    "        ### adding a locus start coordinate and read(or chromosome)name ###\n",
    "        self.loci.append((chrom, position))\n",
    "        \n",
    "    def show_loci(self):\n",
    "        ### show loci in a readable way ###\n",
    "        for chrom, position in self.loci:\n",
    "            print(\"{}\\t{}\".format(chrom, position))\n",
    "            \n",
    "    def load_fastq(path):\n",
    "        ### load and parse fastq file ###\n",
    "        return SeqIO.parse(path, \"fastq\")\n",
    "    \n",
    "    def get_spectrum(path, kmer_size, quality):\n",
    "        ### creates a dictionary of kmers and the loci where they are found ###\n",
    "        file = Kmer_spectrum.load_fastq(path)\n",
    "        kmer_dict = {}\n",
    "        #seq_list = list(islice(file,100))\n",
    "        seq_list = list(file)\n",
    "        for seq_record in seq_list:\n",
    "            seq_lng = len(seq_record.seq)\n",
    "            seq = seq_record.seq\n",
    "            for index in range(seq_lng-kmer_size+1):\n",
    "                current_kmer = seq[index:(index+kmer_size)]\n",
    "                if np.mean(seq_record.letter_annotations[\"phred_quality\"][index:(index+kmer_size)]) >= quality:\n",
    "                    if current_kmer in kmer_dict:\n",
    "                        kmer_dict[current_kmer].increase() #increase counter\n",
    "                    else:\n",
    "                        kmer_dict[current_kmer] = Kmer_spectrum(current_kmer)\n",
    "                    kmer_dict[current_kmer].add_locus(seq_record.name, index)\n",
    "        return kmer_dict\n",
    "    \n",
    "    def get_array(kmer_dict):\n",
    "        ### convert kmer_dict to array of kmer sequence and the frequency of its appearance ###\n",
    "        kmer_array = []\n",
    "        for key in my_dict:\n",
    "            kmer_array.append((str(key), len([x[1] for x in my_dict[key].loci])))\n",
    "        kmer_array = sorted(kmer_array, key=lambda x: x[1], reverse=True)\n",
    "        return kmer_array\n",
    "    \n",
    "    def get_freq_dict(kmer_array):\n",
    "        ### convert kmer array to the dict of frequency and a number of kmers observed with that frequency ###\n",
    "        result = {}\n",
    "        for seq, freq in kmer_array:\n",
    "            if freq not in result:\n",
    "                result[freq] = 1\n",
    "            else:\n",
    "                result[freq] += 1\n",
    "        return result\n",
    "            \n",
    "    def kmer_spectrum_graph(result):\n",
    "        ### creates a picture of kmer distribution, x is a number of kmers, y is an observed frequency, \n",
    "        ### no filtration so far ###\n",
    "        sorted_result = sorted(result.items(), key= lambda x: x[0])\n",
    "        x_axis = [x[0] for x in sorted_result]\n",
    "        y_axis = [x[1] for x in sorted_result]\n",
    "        #plt.axis([0, 100, 0, 70])\n",
    "        plt.bar(x_axis, y_axis)\n",
    "        \n",
    "    def genome_size(freq_dict, kmer_size):\n",
    "        ### calculates genome size without kmers with frequency 1 (considered to be noize) ###\n",
    "        kmer_sum = 0\n",
    "        freq_dict[1] = 0 # filtering kmers with frequency 1 = noize\n",
    "        for freq, kmer_n in freq_dict.items():\n",
    "            kmer_sum += freq*kmer_n\n",
    "        return kmer_sum/kmer_size\n",
    "        \n",
    "            \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "my_dict = Kmer_spectrum.get_spectrum('./test_kmer.fastq', 5, 30)\n",
    "for key in my_dict:\n",
    "    print(key, my_dict[key].loci)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('AAATT', 467)\n",
      "('AAAAA', 385)\n",
      "('AATTT', 384)\n",
      "('TTTTT', 362)\n",
      "('ATTAA', 236)\n",
      "('TTAAA', 227)\n",
      "('AAAAT', 218)\n",
      "('CTTTT', 205)\n",
      "('AAAAG', 204)\n",
      "('TGAAA', 203)\n",
      "('TTAAT', 201)\n",
      "('GAAAT', 193)\n",
      "('TTTAA', 187)\n",
      "('ACATT', 185)\n",
      "('TTTTA', 182)\n",
      "('ATTTG', 181)\n",
      "('AAAGT', 180)\n",
      "('TAAAG', 180)\n",
      "('AATCT', 178)\n",
      "('ATCTT', 175)\n",
      "('AGTTA', 169)\n",
      "('TTGAA', 168)\n",
      "('TTTGA', 166)\n",
      "('TCAGC', 164)\n",
      "('AATTA', 162)\n",
      "('ATCAG', 159)\n",
      "('GACTT', 148)\n",
      "('CAGAA', 145)\n",
      "('GAAAA', 145)\n",
      "('TCAGA', 144)\n",
      "('TTCAA', 143)\n",
      "('TAATT', 143)\n",
      "('GAATC', 141)\n",
      "('CAAAA', 139)\n",
      "('TCAAA', 137)\n",
      "('AGCTT', 137)\n",
      "('AAGCT', 136)\n",
      "('TAAAT', 133)\n",
      "('TTGCA', 132)\n",
      "('AAAGC', 132)\n",
      "('TAAAA', 132)\n",
      "('ATTTA', 132)\n",
      "('TTTAG', 129)\n",
      "('AGAAA', 129)\n",
      "('TGCAA', 129)\n",
      "('AAGTT', 126)\n",
      "('AACTT', 126)\n",
      "('AAAGA', 122)\n",
      "('AATTC', 118)\n",
      "('AAGTC', 116)\n",
      "('ATTCA', 112)\n",
      "('CTAAA', 111)\n",
      "('TTTCA', 109)\n",
      "('GCAAC', 104)\n",
      "('TATAT', 102)\n",
      "('ATTGA', 100)\n",
      "('ACTTG', 100)\n",
      "('TTAAG', 100)\n",
      "('ACTTC', 100)\n",
      "('CTTGC', 100)\n",
      "('CAACT', 99)\n",
      "('TAAGA', 99)\n",
      "('CATTG', 99)\n",
      "('AGACT', 99)\n",
      "('GACAT', 98)\n",
      "('TTCGG', 98)\n",
      "('GTAAA', 98)\n",
      "('CTTCG', 98)\n",
      "('AAGAC', 98)\n",
      "('GGTAA', 96)\n",
      "('AAATC', 94)\n",
      "('CAACA', 94)\n",
      "('TCGGG', 93)\n",
      "('TTGAC', 93)\n",
      "('TGACA', 93)\n",
      "('GCTTT', 92)\n",
      "('TATCA', 92)\n",
      "('TTAGT', 92)\n",
      "('TAGCT', 92)\n",
      "('TCTAG', 92)\n",
      "('TCTTC', 91)\n",
      "('CTAGC', 91)\n",
      "('TTATA', 91)\n",
      "('TAGTA', 91)\n",
      "('AGTAA', 91)\n",
      "('AACAA', 91)\n",
      "('ATATC', 91)\n",
      "('CGGGT', 91)\n",
      "('GGGTA', 91)\n",
      "('CAGCG', 91)\n",
      "('ACAAC', 90)\n",
      "('AGCGA', 90)\n",
      "('GTCAG', 90)\n",
      "('TTCTA', 90)\n",
      "('AACAT', 89)\n",
      "('CTTCT', 89)\n",
      "('GTTAA', 89)\n",
      "('AGTCA', 89)\n",
      "('CGAAT', 88)\n",
      "('GTGAA', 88)\n",
      "('TTTGT', 88)\n",
      "('GCGAA', 87)\n",
      "('GTTAT', 87)\n",
      "('TGTGA', 87)\n",
      "('GTAAC', 87)\n",
      "('TCTTT', 86)\n",
      "('TAACA', 85)\n",
      "('CATTA', 84)\n",
      "('TTGTG', 84)\n",
      "('TCTAA', 76)\n",
      "('ACTTT', 76)\n",
      "('CAGTT', 75)\n",
      "('CAGCA', 74)\n",
      "('AATGT', 74)\n",
      "('GCAGT', 73)\n",
      "('CTCTA', 73)\n",
      "('AAGAT', 72)\n",
      "('AGATT', 72)\n",
      "('TTTTG', 71)\n",
      "('GCTCT', 70)\n",
      "('AGCTC', 70)\n",
      "('ATTCG', 69)\n",
      "('TCATC', 69)\n",
      "('AGCAG', 69)\n",
      "('TTCAT', 68)\n",
      "('GATTC', 66)\n",
      "('CATCA', 66)\n",
      "('TGAAT', 65)\n",
      "('GAATT', 63)\n",
      "('TTCAG', 62)\n",
      "('TGTTT', 60)\n",
      "('GTTTC', 59)\n",
      "('ATTTT', 59)\n",
      "('TCGCT', 58)\n",
      "('TTCGC', 58)\n",
      "('AAGAA', 58)\n",
      "('GCAAA', 57)\n",
      "('TAATG', 57)\n",
      "('AGCTG', 57)\n",
      "('CGCTG', 56)\n",
      "('GGCAA', 56)\n",
      "('GCTGC', 56)\n",
      "('GCTGA', 56)\n",
      "('TGCTG', 55)\n",
      "('CTGTT', 55)\n",
      "('TCGGC', 55)\n",
      "('CGGCA', 55)\n",
      "('ATCGG', 55)\n",
      "('GCTGT', 55)\n",
      "('AATCG', 55)\n",
      "('CTGCT', 54)\n",
      "('CTGAC', 53)\n",
      "('TGTTG', 52)\n",
      "('AGAAT', 52)\n",
      "('TGACT', 51)\n",
      "('CTTCA', 51)\n",
      "('ATGTT', 51)\n",
      "('CTTTA', 49)\n",
      "('GTTGT', 48)\n",
      "('CAAAG', 47)\n",
      "('TTGTT', 47)\n",
      "('GCTTC', 45)\n",
      "('AGAGC', 44)\n",
      "('TTACC', 43)\n",
      "('TGTTA', 43)\n",
      "('GAGCT', 43)\n",
      "('TTTAC', 43)\n",
      "('ATTTC', 42)\n",
      "('TAGAG', 41)\n",
      "('TTAGA', 41)\n",
      "('GTTAC', 40)\n",
      "('TACCC', 39)\n",
      "('TACTA', 39)\n",
      "('TTACT', 39)\n",
      "('GAAGT', 39)\n",
      "('ACTAA', 39)\n",
      "('CCGAA', 38)\n",
      "('CGAAG', 38)\n",
      "('CCCGA', 37)\n",
      "('ACCCG', 36)\n",
      "('GTTGC', 35)\n",
      "('AGTTG', 32)\n",
      "('GCAAG', 29)\n",
      "('CAAGT', 27)\n",
      "('GTCTT', 25)\n",
      "('AGTCT', 25)\n",
      "('GAACT', 23)\n",
      "('AGAAC', 22)\n",
      "('TCAAT', 22)\n",
      "('TCTTA', 22)\n",
      "('CAATG', 21)\n",
      "('CTTAA', 21)\n",
      "('ATGTC', 17)\n",
      "('TATTT', 17)\n",
      "('TTTTC', 16)\n",
      "('TGTCA', 16)\n",
      "('GTCAA', 16)\n",
      "('CAAAT', 14)\n",
      "('ATATT', 13)\n",
      "('TTGTA', 12)\n",
      "('AGCTA', 12)\n",
      "('GTATA', 10)\n",
      "('TTTCT', 10)\n",
      "('ATGAA', 10)\n",
      "('GCTAG', 9)\n",
      "('AGTTT', 9)\n",
      "('TTAAC', 9)\n",
      "('GTTTT', 9)\n",
      "('TTCTG', 8)\n",
      "('CTAGA', 8)\n",
      "('TAGTT', 7)\n",
      "('GATGA', 7)\n",
      "('GGATG', 7)\n",
      "('TAACT', 7)\n",
      "('AGAAG', 7)\n",
      "('GATAT', 7)\n",
      "('TGATA', 7)\n",
      "('TAGAA', 7)\n",
      "('TTATT', 6)\n",
      "('TTCGA', 6)\n",
      "('TGGAT', 6)\n",
      "('AGCAA', 6)\n",
      "('GAAGA', 6)\n",
      "('TCTGA', 6)\n",
      "('CTGAT', 6)\n",
      "('GATTT', 6)\n",
      "('TGTAT', 6)\n",
      "('GTAGT', 5)\n",
      "('CATTT', 5)\n",
      "('ATTAG', 5)\n",
      "('CGACA', 5)\n",
      "('ACGAA', 5)\n",
      "('CAGGT', 5)\n",
      "('TCAGG', 5)\n",
      "('CATCT', 5)\n",
      "('AACGA', 5)\n",
      "('ATATA', 5)\n",
      "('TCTGC', 5)\n",
      "('TCGAC', 5)\n",
      "('TAACG', 5)\n",
      "('AGGTA', 5)\n",
      "('GGAAT', 4)\n",
      "('CTTGT', 4)\n",
      "('CGGAA', 4)\n",
      "('ATGGA', 4)\n",
      "('TGCAG', 4)\n",
      "('ATCTG', 4)\n",
      "('TCGGA', 4)\n",
      "('CTGCA', 4)\n",
      "('CTGCC', 4)\n",
      "('ATGTG', 4)\n",
      "('TGCCA', 4)\n",
      "('CTGAA', 4)\n",
      "('TCTTG', 4)\n",
      "('TGTAG', 4)\n",
      "('TTTAT', 4)\n",
      "('AATGA', 3)\n",
      "('GTTTA', 3)\n",
      "('CGTTA', 3)\n",
      "('CGAAA', 3)\n",
      "('TCGTT', 3)\n",
      "('TATAA', 3)\n",
      "('TTTGG', 3)\n",
      "('GCCAT', 3)\n",
      "('AATAA', 3)\n",
      "('CCATT', 3)\n",
      "('TTCGT', 3)\n",
      "('CTTAT', 3)\n",
      "('TATTG', 3)\n",
      "('CTTTG', 2)\n",
      "('CTAAT', 2)\n",
      "('CAAGA', 2)\n",
      "('TACCT', 2)\n",
      "('TGAAG', 2)\n",
      "('AGTAT', 2)\n",
      "('ATTAT', 2)\n",
      "('TTGCC', 2)\n",
      "('GTTCA', 2)\n",
      "('TAATA', 2)\n",
      "('CCAAA', 2)\n",
      "('TAAGC', 2)\n",
      "('ATAAT', 2)\n",
      "('CTTTC', 2)\n",
      "('ATCTA', 2)\n",
      "('AGTTC', 2)\n",
      "('ACTCT', 2)\n",
      "('AAATG', 2)\n",
      "('TGGTA', 2)\n",
      "('TGTAA', 2)\n",
      "('ATAAA', 2)\n",
      "('ACCTG', 2)\n",
      "('CCTGA', 2)\n",
      "('TTGGT', 2)\n",
      "('ATTGT', 2)\n",
      "('AACTC', 2)\n",
      "('TCGAA', 2)\n",
      "('TTGAT', 2)\n",
      "('AATGG', 2)\n",
      "('ATAAC', 2)\n",
      "('CTATA', 1)\n",
      "('GGACT', 1)\n",
      "('GGGCA', 1)\n",
      "('ACAGT', 1)\n",
      "('GTCGA', 1)\n",
      "('ACATA', 1)\n",
      "('GTGGA', 1)\n",
      "('CCGTT', 1)\n",
      "('TGCCG', 1)\n",
      "('TTAGC', 1)\n",
      "('GCCCG', 1)\n",
      "('ACCAA', 1)\n",
      "('TAGAT', 1)\n",
      "('TTTGC', 1)\n",
      "('GCAAT', 1)\n",
      "('GCCGT', 1)\n",
      "('ACTTA', 1)\n",
      "('CATAA', 1)\n",
      "('AGGAC', 1)\n",
      "('CAATA', 1)\n",
      "('TACTT', 1)\n",
      "('GCCAA', 1)\n",
      "('AGAGT', 1)\n",
      "('TTAGG', 1)\n",
      "('TGCCC', 1)\n",
      "('AAGCG', 1)\n",
      "('TCACA', 1)\n",
      "('GATAA', 1)\n",
      "('TAGGA', 1)\n",
      "('CACAA', 1)\n",
      "('TTGGA', 1)\n",
      "('AAGAG', 1)\n",
      "('AACAG', 1)\n",
      "('TGTCG', 1)\n",
      "('TCATT', 1)\n",
      "('CGGGC', 1)\n",
      "('CGATT', 1)\n",
      "('TTATC', 1)\n",
      "('CGTTT', 1)\n",
      "('GAGTA', 1)\n",
      "('AATCA', 1)\n",
      "('TGTGG', 1)\n",
      "('AGCGT', 1)\n",
      "('GCTAT', 1)\n",
      "('CTAGT', 1)\n",
      "('ATCAT', 1)\n",
      "('GAAAG', 1)\n",
      "('ATAAG', 1)\n",
      "('TATTA', 1)\n",
      "('TCAAG', 1)\n",
      "('AAACG', 1)\n",
      "('AACTA', 1)\n",
      "('TTCAC', 1)\n",
      "('AGTAG', 1)\n",
      "('TTCTT', 1)\n",
      "('CTCAA', 1)\n",
      "('TGAAC', 1)\n",
      "('GCGTA', 1)\n",
      "('CAATT', 1)\n",
      "('TACCA', 1)\n",
      "('CAGTA', 1)\n",
      "('CGTAT', 1)\n",
      "('TTGCT', 1)\n",
      "('ATCAA', 1)\n",
      "('CCTTC', 1)\n",
      "('ACATC', 1)\n",
      "('GCGAT', 1)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/nastya/anaconda3/lib/python3.5/site-packages/Bio/Seq.py:155: BiopythonWarning: Biopython Seq objects now use string comparison. Older versions of Biopython used object comparison. During this transition, please use hash(id(my_seq)) or my_dict[id(my_seq)] if you want the old behaviour, or use hash(str(my_seq)) or my_dict[str(my_seq)] for the new string hashing behaviour.\n",
      "  \"the new string hashing behaviour.\", BiopythonWarning)\n"
     ]
    }
   ],
   "source": [
    "my_array = Kmer_spectrum.get_array(my_dict)\n",
    "for kmer in my_array:\n",
    "    print(kmer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "result = Kmer_spectrum.get_freq_dict(my_array)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXoAAAD8CAYAAAB5Pm/hAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMS4wLCBo\ndHRwOi8vbWF0cGxvdGxpYi5vcmcvpW3flQAAD7lJREFUeJzt3V2MXGd9x/HvrzHhxSDs0K3lJkR2\nRZQoRUpCVyEIhNqEVIEg7IsoCkJ0VbnyDbShRQLTXiH1wkgVkEoVkpUA24qGpCbUViLRpiYIVWoD\na5JCEqd1kjrFrl8WiIFyUQj8ezHHeGu87Oy87K6f+X6k0ZznOefs+fv42Z/PPHNmnKpCktSuX1nt\nAiRJ42XQS1LjDHpJapxBL0mNM+glqXEGvSQ1bsmgT3JlkscXPH6Q5ANJLknycJLD3fPGlShYkrQ8\nWc599EkuAo4BbwTeB3yvqnYn2QVsrKoPj6dMSdKgljt1cxPwbFU9D2wDZrv+WWD7KAuTJI3GumVu\nfwdwb7e8qaqOd8sngE3n2yHJTmAnwPr163/rqquuGqROSZpYBw8e/E5VTQ26f99TN0kuBv4b+M2q\nOpnkdFVtWLD+har6pfP009PTNTc3N2itkjSRkhysqulB91/O1M3bgW9U1cmufTLJ5q6IzcCpQYuQ\nJI3PcoL+3ZydtgHYD8x0yzPAvlEVJUkanb6CPsl64GbggQXdu4GbkxwG3ta1JUlrTF9vxlbVj4DX\nnNP3XXp34UiS1jA/GStJjTPoJalxBr0kNc6gl6TGLfeTsSOxZddDP18+svvW1ShBkiaGV/SS1DiD\nXpIaZ9BLUuMMeklqnEEvSY0z6CWpcQa9JDXOoJekxhn0ktQ4g16SGmfQS1LjDHpJapxBL0mNM+gl\nqXEGvSQ1zqCXpMYZ9JLUuL6CPsmGJHuTPJ3kUJI3JbkkycNJDnfPG8ddrCRp+fq9or8L+FJVXQVc\nAxwCdgEHquoK4EDXliStMUsGfZJXA28F7gGoqh9X1WlgGzDbbTYLbB9XkZKkwfVzRb8VmAc+k+Sx\nJHcnWQ9sqqrj3TYngE3n2znJziRzSebm5+dHU7UkqW/9BP064A3Ap6rqOuBHnDNNU1UF1Pl2rqo9\nVTVdVdNTU1PD1itJWqZ+gv4ocLSqHu3ae+kF/8kkmwG651PjKVGSNIwlg76qTgDfTnJl13UT8BSw\nH5jp+maAfWOpUJI0lHV9bveHwOeSXAw8B/w+vX8k7k+yA3geuH08JUqShtFX0FfV48D0eVbdNNpy\nJEmj5idjJalxBr0kNc6gl6TGGfSS1DiDXpIaZ9BLUuMMeklqnEEvSY0z6CWpcQa9JDXOoJekxhn0\nktQ4g16SGmfQS1LjDHpJapxBL0mNM+glqXEGvSQ1zqCXpMYZ9JLUOINekhpn0EtS49b1s1GSI8AP\ngZ8CL1bVdJJLgPuALcAR4PaqemE8ZUqSBrWcK/rfqaprq2q6a+8CDlTVFcCBri1JWmOGmbrZBsx2\ny7PA9uHLkSSNWr9BX8A/JjmYZGfXt6mqjnfLJ4BN59sxyc4kc0nm5ufnhyxXkrRcfc3RA2+pqmNJ\nfg14OMnTC1dWVSWp8+1YVXuAPQDT09Pn3UaSND59XdFX1bHu+RTwReB64GSSzQDd86lxFSlJGtyS\nQZ9kfZJXnVkGfhd4AtgPzHSbzQD7xlWkJGlw/UzdbAK+mOTM9n9bVV9K8nXg/iQ7gOeB28dXpiRp\nUEsGfVU9B1xznv7vAjeNoyhJ0uj4yVhJapxBL0mNM+glqXEGvSQ1zqCXpMYZ9JLUOINekhpn0EtS\n4wx6SWqcQS9JjTPoJalxBr0kNc6gl6TGGfSS1DiDXpIaZ9BLUuMMeklqnEEvSY0z6CWpcQa9JDXO\noJekxhn0ktS4voM+yUVJHkvyYNfemuTRJM8kuS/JxeMrU5I0qOVc0d8JHFrQ/hjwiap6HfACsGOU\nhUmSRqOvoE9yGXArcHfXDnAjsLfbZBbYPo4CJUnD6feK/pPAh4Cfde3XAKer6sWufRS4dMS1SZJG\nYMmgT/JO4FRVHRzkAEl2JplLMjc/Pz/Ij5AkDaGfK/o3A+9KcgT4PL0pm7uADUnWddtcBhw7385V\ntaeqpqtqempqagQlS5KWY8mgr6qPVNVlVbUFuAP4clW9B3gEuK3bbAbYN7YqJUkDG+Y++g8Df5Lk\nGXpz9veMpiRJ0iitW3qTs6rqK8BXuuXngOtHX5IkaZT8ZKwkNc6gl6TGGfSS1DiDXpIaZ9BLUuMM\neklqnEEvSY0z6CWpcQa9JDXOoJekxhn0ktQ4g16SGmfQS1LjDHpJapxBL0mNM+glqXEGvSQ1zqCX\npMYZ9JLUOINekhpn0EtS4wx6SWrckkGf5GVJvpbk35I8meSjXf/WJI8meSbJfUkuHn+5kqTl6ueK\n/n+BG6vqGuBa4JYkNwAfAz5RVa8DXgB2jK9MSdKglgz66vmfrvmS7lHAjcDern8W2D6WCiVJQ+lr\njj7JRUkeB04BDwPPAqer6sVuk6PApeMpUZI0jHX9bFRVPwWuTbIB+CJwVb8HSLIT2Alw+eWX/8L6\nLbse+vnykd239vtjJUl9WtZdN1V1GngEeBOwIcmZfyguA44tss+eqpququmpqamhipUkLV8/d91M\ndVfyJHk5cDNwiF7g39ZtNgPsG1eRkqTB9TN1sxmYTXIRvX8Y7q+qB5M8BXw+yZ8DjwH3jLFOSdKA\nlgz6qvomcN15+p8Drh9HUZKk0fGTsZLUOINekhpn0EtS4wx6SWqcQS9JjTPoJalxBr0kNc6gl6TG\nGfSS1DiDXpIaZ9BLUuMMeklqnEEvSY0z6CWpcQa9JDXOoJekxhn0ktQ4g16SGmfQS1LjDHpJapxB\nL0mNM+glqXEGvSQ1bsmgT/LaJI8keSrJk0nu7PovSfJwksPd88bxlytJWq5+ruhfBD5YVVcDNwDv\nS3I1sAs4UFVXAAe6tiRpjVky6KvqeFV9o1v+IXAIuBTYBsx2m80C28dVpCRpcMuao0+yBbgOeBTY\nVFXHu1UngE2L7LMzyVySufn5+SFKlSQNou+gT/JK4AvAB6rqBwvXVVUBdb79qmpPVU1X1fTU1NRQ\nxUqSlq+voE/yEnoh/7mqeqDrPplkc7d+M3BqPCVKkobRz103Ae4BDlXVxxes2g/MdMszwL7RlydJ\nGta6PrZ5M/Be4FtJHu/6/hTYDdyfZAfwPHD7eEqUJA1jyaCvqn8Gssjqm0ZbjiRp1PxkrCQ1zqCX\npMYZ9JLUOINekhpn0EtS4wx6SWqcQS9JjTPoJalxBr0kNa6fr0BYNVt2PfTz5SO7b13FSiTpwuUV\nvSQ1zqCXpMat6ambhZzGkaTBeEUvSY0z6CWpcWsu6Lfseuj/TdNIkoaz5oJekjRaBr0kNc6gl6TG\nGfSS1DiDXpIaZ9BLUuOWDPokn05yKskTC/ouSfJwksPd88bxlilJGlQ/V/SfBW45p28XcKCqrgAO\ndG1J0hq0ZNBX1VeB753TvQ2Y7ZZnge0jrkuSNCKDfqnZpqo63i2fADYttmGSncBOgMsvv3zAw0nS\n6rnQv1Rx6Ddjq6qA+iXr91TVdFVNT01NDXs4SdIyDRr0J5NsBuieT42uJEnSKA0a9PuBmW55Btg3\nmnL655efSVJ/+rm98l7gX4ArkxxNsgPYDdyc5DDwtq4tSVqDlnwztqrevciqm0ZciyRpDC6Y/0rw\nl7nQ3xGXpHHyKxAkqXEGvSQ1zqCXRsC7wLSWGfSS1DiDXpIaZ9BLUuMMeklqnEEvSY1r4gNTunD4\n4Ta14Mw4HtUYXuznjepOLq/oJalxBr0kNa75qZtRv8TSyrvQpnsutHonTT/TJEd237pq2TGOD955\nRS9JjTPoJalxzU/dLDTql9Rr/edp+cb9d7rWphInfcyNYprkQjiHXtFLUuMMeklqnEEvSY1rbo5+\n0PmyYedOV/K4F8Kc4EL9/hnH9WnDYX7mOOfUx/n3OOx4PN9+C8/FYnPbS92yOE5r4f2Ptfq76RW9\nJDXOoJekxg01dZPkFuAu4CLg7qraPZKqVlm/L0tXqoZxvPTu52csZ7tBzlk/tS7nuMs51rBTZguN\n+9baQaZJ+j3WoDX1s99amUJZaK3UtNJ1DHxFn+Qi4K+AtwNXA+9OcvWoCpMkjcYwUzfXA89U1XNV\n9WPg88C20ZQlSRqVVNVgOya3AbdU1R907fcCb6yq95+z3U5gZ9d8PfDE4OU25VeB76x2EWuE5+Is\nz8VZnouzrqyqVw2689hvr6yqPcAegCRzVTU97mNeCDwXZ3kuzvJcnOW5OCvJ3DD7DzN1cwx47YL2\nZV2fJGkNGSbovw5ckWRrkouBO4D9oylLkjQqA0/dVNWLSd4P/AO92ys/XVVPLrHbnkGP1yDPxVme\ni7M8F2d5Ls4a6lwM/GasJOnC4CdjJalxBr0kNW5Fgj7JLUn+PckzSXatxDHXiiSvTfJIkqeSPJnk\nzq7/kiQPJzncPW9c7VpXSpKLkjyW5MGuvTXJo934uK97c795STYk2Zvk6SSHkrxpUsdFkj/ufj+e\nSHJvkpdNyrhI8ukkp5I8saDvvOMgPX/ZnZNvJnlDP8cYe9D7VQm8CHywqq4GbgDe1/35dwEHquoK\n4EDXnhR3AocWtD8GfKKqXge8AOxYlapW3l3Al6rqKuAaeudk4sZFkkuBPwKmq+r19G7uuIPJGRef\nBW45p2+xcfB24IrusRP4VD8HWIkr+on+qoSqOl5V3+iWf0jvl/lSeudgtttsFti+OhWurCSXAbcC\nd3ftADcCe7tNJuJcJHk18FbgHoCq+nFVnWZCxwW9OwBfnmQd8ArgOBMyLqrqq8D3zulebBxsA/66\nev4V2JBk81LHWImgvxT49oL20a5v4iTZAlwHPApsqqrj3aoTwKZVKmulfRL4EPCzrv0a4HRVvdi1\nJ2V8bAXmgc9001h3J1nPBI6LqjoG/AXwX/QC/vvAQSZzXJyx2DgYKE99M3aFJHkl8AXgA1X1g4Xr\nqnePa/P3uSZ5J3Cqqg6udi1rwDrgDcCnquo64EecM00zQeNiI70r1a3ArwPr+cWpjIk1inGwEkE/\n8V+VkOQl9EL+c1X1QNd98sxLru751GrVt4LeDLwryRF6U3g30pun3tC9ZIfJGR9HgaNV9WjX3ksv\n+CdxXLwN+M+qmq+qnwAP0BsrkzguzlhsHAyUpysR9BP9VQndHPQ9wKGq+viCVfuBmW55Bti30rWt\ntKr6SFVdVlVb6I2DL1fVe4BHgNu6zSblXJwAvp3kyq7rJuApJnBc0JuyuSHJK7rflzPnYuLGxQKL\njYP9wO91d9/cAHx/wRTP4qpq7A/gHcB/AM8Cf7YSx1wrD+At9F52fRN4vHu8g97c9AHgMPBPwCWr\nXesKn5ffBh7sln8D+BrwDPB3wEtXu74VOgfXAnPd2Ph7YOOkjgvgo8DT9L7G/G+Al07KuADupffe\nxE/ovdLbsdg4AELvLsZngW/Ru1NpyWP4FQiS1DjfjJWkxhn0ktQ4g16SGmfQS1LjDHpJapxBL0mN\nM+glqXH/B8EQRMh7QsZ0AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7fbce80e5ef0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Kmer_spectrum.kmer_spectrum_graph(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda env:anaconda3]",
   "language": "python",
   "name": "conda-env-anaconda3-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
